lib/web/users.go

```diff
@@ -169,7 +169,13 @@ func updateUser(r *http.Request, m userAPIGetter) (*ui.User, error) {
 
 	user.SetRoles(req.Roles)
 
-	updateUserTraits(req, user)
+	// we do not want situation where one trait set (allTraits)
+	// overrides another (traits).
+	if len(req.AllTraits) > 0 {
+		user.SetTraits(req.AllTraits)
+	} else {
+		updateUserTraits(req, user)
+	}
```

Comment:
This PR makes use of `AllTraits` field to manage user Traits. 

I am guessing we would ideally want to either utilize `AllTraits` or `Traits` (which just selectively updates specific traits.  But tentatively tagging @marcoandredinis as Github shows you added user traits editing functionality in user API. Did I miss anything here?

Reply:
From what I can see, Traits contains only the default used traits (logins, aws arn roles, windows logins, db users, db names, ...).
While AllTraits contains everything (I think it was introduced as part of Access Lists).

I don't have the Access List context, but given that Lisa and Roman are already your reviewers, I think you are good.